subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -79,6 +79,9 @@ static struct hawkbit_config {
 	int32_t action_id;
 #ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
 	char server_addr[DNS_MAX_NAME_SIZE + 1];
+#ifdef CONFIG_HAWKBIT_USE_DOMAIN_NAME
+	char server_domain[CONFIG_DNS_RESOLVER_MAX_QUERY_LEN + 1];
```

Comment:
Same comment as in https://github.com/zephyrproject-rtos/zephyr/pull/89497#discussion_r2074960577 so it's not overlooked here. One option could be to define a new Kconfig for the buffer size for HawkBit, defaulting to DNS_RESOLVER_MAX_QUERY_LEN if DNS_RESOLVER is enabled.

Reply:
What should be the default value if DNS_RESOLVER_MAX_QUERY_LEN is undefined?

For server_addr, the length can default to INET6_ADDRSTRLEN to support an IP address. However here it must support a DNS name. Is it okay to hardcode 255 as the default when there are no Kconfigs available?

for example:
```
#ifdef CONFIG_DNS_RESOLVER_MAX_QUERY_LEN
#define SERVER_DOMAIN_LEN CONFIG_DNS_RESOLVER_MAX_QUERY_LEN
#else¬
#define SERVER_DOMAIN_LEN 255
#endif¬
 ```

Reply:
That's one option to have the default hardcoded as you've posted, or it can also be resolved via Kconfig if a dedicated config for the buffer lengths was created:
```
default DNS_RESOLVER_MAX_QUERY_LEN if DNS_RESOLVER 
default 255
```

Reply:
I implemented a Kconfig for the domain name buffer length.